sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -440,8 +462,13 @@ def _fit(self, X, y, incremental=False):
         )
 
         X, y = self._validate_input(X, y, incremental, reset=first_pass)
+        # Handle sample_weight
+        if sample_weight is not None:
+            sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
+            if sum(sample_weight != 0) == 0:
```

Comment:
Couldn't we just check if the sum = 0? No need for != 0.

Reply:
Thank you for the comment. There could be a case where the sum is 0 because there are negative weights, but sample_weight is supposed to be non-negative. I will update it together with setting `ensure_non_negative=True` for `_check_sample_weight()`.